File: app/src/main/java/amrabed/android/release/evaluation/preferences/Preferences.java
package amrabed.android.release.evaluation.preferences;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

import amrabed.android.release.evaluation.R;

/**
 * Provides access to shared preferences globally
 */

public class Preferences {
    private Preferences() {
    }


    public static int getFastingDays(Context context) {
        return getSharedPreferences(context).getInt("fastingDays", 0);
    }

    public static long getLastDayOfFasting(Context context) {
        return getSharedPreferences(context).getLong("ldof", 0);
    }

    public static void setLastDayOfFasting(Context context, long date) {
        getSharedPreferences(context).edit().putLong("ldof", date).apply();
    }

    public static void removeLastDayOfFasting(Context context) {
        getSharedPreferences(context).edit().remove("ldof").apply();
    }

    public static String[] getDefaultTaskTitles(Context context) {
        return isMale(context) ? context.getResources().getStringArray(R.array.m_activities) :
                context.getResources().getStringArray(R.array.f_activities);
    }

    private static boolean isMale(Context context) {
        return getSharedPreferences(context).getBoolean("gender", true);
    }

    private static SharedPreferences getSharedPreferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static long getLastAddedDay(Context context) {
        return getSharedPreferences(context).getLong("last added day", 0);
    }

    public static void setLastAddedDay(Context context, long day) {
        getSharedPreferences(context).edit().putLong("last added day", day).apply();
    }
}

File: app/src/main/java/amrabed/android/release/evaluation/utilities/preferences/Preferences.kt
package amrabed.android.release.evaluation.utilities.preferences

import amrabed.android.release.evaluation.R
import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

/**
 * Provides access to shared preferences globally
 */
object Preferences {
    fun getFastingDays(context: Context): Int {
        return getSharedPreferences(context).getInt("fastingDays", 0)
    }

    fun getLastDayOfFasting(context: Context): Long {
        return getSharedPreferences(context).getLong("ldof", 0)
    }

    fun setLastDayOfFasting(context: Context, date: Long) {
        getSharedPreferences(context).edit().putLong("ldof", date).apply()
    }

    fun removeLastDayOfFasting(context: Context) {
        getSharedPreferences(context).edit().remove("ldof").apply()
    }

    fun getDefaultTaskTitles(context: Context): Array<String> {
        return if (isMale(context)) context.resources.getStringArray(R.array.mActivities) else context.resources.getStringArray(
            R.array.fActivities
        )
    }

    private fun isMale(context: Context): Boolean {
        return getSharedPreferences(context).getBoolean("gender", true)
    }

    private fun getSharedPreferences(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

    fun isHijriCalendar(context: Context): Boolean {
        return getSharedPreferences(context).getBoolean("calendar", true)
    }
}